osbrain/tests/test_agent_ipc_sockets.py

```diff
@@ -0,0 +1,74 @@
+"""
+Test file for everything IPC socket related
+"""
+import pytest
+
+from osbrain import run_agent
+from osbrain import run_nameserver
+
+from common import nsproxy  # pragma: no flakes
+
+
+def test_agent_close_ipc_socket_agent_shutdown(nsproxy):
+    """
+    Check that the socket is closed and the socket file removed after the agent
+    is shut down.
+    """
+    agent = run_agent('name')
+    address = agent.bind('PUSH')
+    agent.shutdown()
+    assert not address.address.exists()
+
+
+def test_agent_close_ipc_socket_agent_kill(nsproxy):
+    """
+    Check that the socket is closed and the socket file removed after the agent
+    is killed.
+    """
+    agent = run_agent('name')
+    address = agent.bind('PUSH')
+    agent.kill()
+    assert not address.address.exists()
+
+
+def test_agent_close_ipc_socket_agent_blocked(nsproxy):
+    """
+    Check that the socket is closed and the socket file removed when a blocked
+    agent is requested to shut down.
+    """
+    def block(agent):
+        agent.send('out', 'blocking...')
+
+    blocker = run_agent('blocker')
+    blocker.set_method(block)
+    addr = blocker.bind('PUSH', alias='out')
+    blocker.after(0, 'block')
+    nsproxy.shutdown(timeout=1.)
+
+    assert not addr.address.exists()
+
+
+def test_agent_close_ipc_socket_agent_crash():
+    """
+    Check that the socket is closed and the socket file removed after the agent
+    crashes.
+    """
```

Comment:
I would rather call this `test_agent_close_ipc_socket_agent_crash_nameserver_shutdown`. And maybe in the docstring: "[...] after the agent crashes and the name server calls for shutdown.".

What you think?

Reply:
Maybe we can do the same with the test just above (to specify that the shutdown call is executed from the name server).

Reply:
:joy: